bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/cache/ExpiringCacheAsync.java

```diff
@@ -38,14 +39,24 @@
     /**
      * Create a new instance.
      *
-     * @param expiry the duration in milliseconds for how long the value stays valid. Must be greater than 0.
+     * @param expiry the duration in milliseconds for how long the value stays valid. Must be positive.
      * @throws IllegalArgumentException For an expire value <=0.
      */
-    public ExpiringCacheAsync(long expiry) throws IllegalArgumentException {
-        if (expiry <= 0) {
+    public ExpiringCacheAsync(Duration expiry) {
+        if (expiry.isNegative() || expiry.isZero()) {
```

Comment:
Shall we add this check to the `ExpiringCache` constructor as well?

Reply:
Its unlikely that someone constructs it like, but its a framework so th additional robustness makes sense

Reply:
https://github.com/openhab/openhab-core/pull/1209 - one year later ... :rofl:

Reply:
😁